pkg/storegateway/store_gateway_client.go

```diff
@@ -47,32 +47,32 @@ func dialStoreGatewayClient(clientCfg grpcclient.Config, addr string, requestDur
 		return nil, errors.Wrapf(err, "failed to dial store-gateway %s", addr)
 	}
 
-	return &storeGatewayClient{
+	return &ClientImpl{
 		StoreGatewayClient: storegatewaypb.NewStoreGatewayClient(conn),
 		HealthClient:       grpc_health_v1.NewHealthClient(conn),
 		conn:               conn,
 	}, nil
 }
 
-type storeGatewayClient struct {
+type ClientImpl struct {
 	storegatewaypb.StoreGatewayClient
 	grpc_health_v1.HealthClient
 	conn *grpc.ClientConn
 }
 
-func (c *storeGatewayClient) Close() error {
+func (c *ClientImpl) Close() error {
```

Comment:
`*Impl` feels like kind of a Java-ism. Doesn't seem like we use it anywhere else in the Mimir codebase.